maven-core/src/main/java/org/apache/maven/DefaultMaven.java

```diff
@@ -367,11 +354,37 @@ private MavenExecutionResult doExecute( MavenExecutionRequest request, MavenSess
         return result;
     }
 
-    private void afterSessionStart( MavenSession session )
+    private void finishRepositorySystemSession( MavenSession session, DefaultRepositorySystemSession repoSession )
```

Comment:
The name of this method sounds a bit odd to me. Seems more like it "adds workspace readers" or something like that, than that it "finishes" something. But maybe in the context of this change, it makes sense?

Reply:
I just found that in maven 4 (in contrast to 3.8.x) everything is extracted into a method, if you look a bit down the line, there is the `RepositorySystemSession#setReadOnly()` so I thought this might contain every action before the session is made readonly/finalized, but of course I would choose another name if it is appropriate, so let me know!

Reply:
done